webserver/webserver/src/main/java/io/helidon/webserver/ConfiguredTlsManager.java

```diff
@@ -0,0 +1,278 @@
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+
+import io.helidon.common.LazyValue;
+import io.helidon.common.pki.KeyConfig;
+
+/**
+ * The default configured {@link TlsManager} implementation.
+ */
+public class ConfiguredTlsManager implements TlsManager {
+    // secure random cannot be stored in native image, it must
+    // be initialized at runtime
+    private static final LazyValue<SecureRandom> RANDOM = LazyValue.create(SecureRandom::new);
+
+    private final String name;
+    private final String type;
+    private final Set<Consumer<SSLContext>> subscribers = new LinkedHashSet<>();
+    private volatile SSLContext sslContext;
+
+    ConfiguredTlsManager() {
+        this("@default", "tls-manager");
+    }
+
+    /**
+     * Configured tls manager constructor.
+     *
+     * @param name the manager name
+     * @param type the manager type
+     */
+    protected ConfiguredTlsManager(String name,
+                                   String type) {
+        this.name = Objects.requireNonNull(name);
+        this.type = Objects.requireNonNull(type);
+    }
+
+    /**
+     * The manager name.
+     *
+     * @return manager name
+     */
+    // @Override // NamedService
+    public String name() {
+        return name;
+    }
+
+    /**
+     * The manager type.
+     *
+     * @return manager type
+     */
+    // @Override // NamedService
+    public String type() {
+        return type;
+    }
+
+    @Override // TlsManager
+    public SSLContext sslContext() {
+        return sslContext;
+    }
+
+    @Override // TlsManager
+    public void subscribe(Consumer<SSLContext> sslContextConsumer) {
+        subscribers.add(Objects.requireNonNull(sslContextConsumer));
+    }
+
+    @Override // TlsManager
+    public void init(WebServerTls tlsConfig) {
+        SSLContext explicitSslContext = tlsConfig.explicitSslContext().orElse(null);
+        if (explicitSslContext != null) {
+            this.sslContext = explicitSslContext;
+            return;
+        }
+
+        if (tlsConfig.privateKeyConfig() == null) {
+            throw new IllegalStateException("Private key must be configured when SSL is enabled.");
+        }
+
+        try {
+            KeyManagerFactory kmf = buildKmf(tlsConfig.privateKeyConfig());
+            TrustManagerFactory tmf = buildTmf(tlsConfig);
+
+            initSslContext(tlsConfig,
+                           kmf.getKeyManagers(),
+                           tmf.getTrustManagers());
+        } catch (IOException | GeneralSecurityException e) {
+            throw new IllegalStateException("Failed to build server SSL Context!", e);
+        }
+    }
+
+    /**
+     * Initialize and set the SSL context given the provided configuration.
+     *
+     * @param tlsConfig     the tls config
+     * @param keyManagers   the key managers
+     * @param trustManagers the trust managers
+     * @deprecated this method will removed in a future release.
```

Comment:
I think you are missing "be" -> will be removed

Reply:
pushed